airbyte-integrations/connectors/destination-amazon-sqs/destination_amazon_sqs/destination.py

```diff
@@ -0,0 +1,171 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+
+import json
+from typing import Any, Iterable, Mapping
+from uuid import uuid4
+
+import boto3
+from airbyte_cdk import AirbyteLogger
+from airbyte_cdk.destinations import Destination
+from airbyte_cdk.models import AirbyteConnectionStatus, AirbyteMessage, ConfiguredAirbyteCatalog, Status, Type
+from botocore.exceptions import ClientError
+
+
+class DestinationAmazonSqs(Destination):
+    def queue_is_fifo(self, url: str) -> bool:
+        return url.endswith(".fifo")
+
+    def parse_queue_name(self, url: str) -> str:
+        return url.rsplit("/", 1)[-1]
+
+    def send_single_message(self, queue, message) -> dict:
+        return queue.send_message(**message)
+
+    def build_sqs_message(self, record, message_body_key=None):
+        data = None
+        if message_body_key:
+            data = record.data.get(message_body_key)
+            if data is None:
+                raise Exception("Message had no attribute of the configured Message Body Key: " + message_body_key)
+        else:
+            data = json.dumps(record.data)
+
+        message = {"MessageBody": data}
+
+        return message
+
+    def add_attributes_to_message(self, record, message):
+        attributes = {"airbyte_emitted_at": {"StringValue": str(record.emitted_at), "DataType": "String"}}
+        message["MessageAttributes"] = attributes
+        return message
+
+    def set_message_delay(self, message, message_delay):
+        message["DelaySeconds"] = message_delay
+        return message
+
+    def set_message_fifo_properties(self, message, message_group_id, use_content_dedupe=False):
```

Comment:
can you add the documentation link to have better picture on why we need these properties ?

Reply:
Added documentation link and comment. These are required properties specifically when using FIFO queues